cypress/integration/cases.test.ts

```diff
@@ -0,0 +1,67 @@
+export default context("Cases", () => {
+  const apiFixture = {
+    startdatum: "2020-02-02",
+    einddatum: "2019-01-01",
+    zaaktype: "",
+    toelichting: "API boulevard 12, 9999ZZ",
+    omschrijving: "Already added"
+  }
+
+  const browserFixture = {
+    startdatum: "2019-01-01",
+    einddatum:  "2019-02-03",
+    zaaktype: "Illegale vakantieverhuur",
+    toelichting: "Teststraat 1, 1111AA",
+    omschrijving: "End to end test"
+  }
+
+  beforeEach(() => {
+    cy.cleanDatabase()
+      .then((response: any) => {
+        // After cleaning the database, a new case_type URL is generated.
+        // We need it to be able to programmatically create a case
+        apiFixture.zaaktype = response.body.case_type.url
+      })
+  })
+
+  it("should be able to add a case", () => {
+    cy.visit("cases/create")
+    cy.autoFill(browserFixture)
+    cy.getSubmitButton().click()
+
+    // Assertions:
+    // ---
+
+    cy.get("[data-e2e-id='alert'][data-e2e-variant='success']").should("be.visible")
+  })
+
+  it("should be able to edit a case", () => {
+    // We need a pre-existing case to edit.
+    cy.postToAPI("cases/", apiFixture)
```

Comment:
No `async` + `await`?

Reply:
No unfortunately, Cypress does not return real promises.
https://github.com/cypress-io/cypress/issues/1417

Reply:
Ok. That's quite ugly. Also seems it will never get supported.